components/pill/index.jsx

```diff
@@ -198,7 +198,7 @@ class Pill extends React.Component {
 	};
 
 	renderIcon = () => {
-		const icon = this.props.icon || this.props.avatar;
+		const icon = this.props?.icon || this.props?.avatar;
```

Comment:
What is the use case for `this.props` not being defined?

Reply:
Actually the undefined problem is comming beacuse of selected-listbox.jsx getIcon funtion as In some cases option only have icon or in other case only avatar, there i have applied const iconObject = option?.icon || null; and here const icon = this.props?.icon || this.props?.avatar; for the extra safety.